### PR TITLE
🎨 Update docs theme and add tutorial

### PR DESCRIPTION
**Pull request by chrisjsewell**
- Update the theme to furo and colors to match new branding
  - branding coloring loosely based on https://mintlify.com/docs/quickstart 
- Restructure toc-trees, to make the documentation easier to navigate
- Improve landing page
   - Improve top "hero" section
   - Add feature section
   - move examples to tutorial
- Add tutorial page, with worked example covering most directives
  - took some things from https://sphinx-needs-demo.readthedocs.io 

See: https://sphinx-needs--1184.org.readthedocs.build/en/1184/

### CODE REVIEW

File: docs/tutorial.rst [hunk 1]
```diff
@@ -0,0 +1,346 @@
...
+you can explore the full range of options available in sphinx-needs by reading the rest of the documentation.
+
+For a more complex project example, check out the `sphinx-needs-demo <https://sphinx-needs-demo.readthedocs.io>`_ site.
+
+Also, see :ref:`other extensions <other-extensions>` offered by UseBlocks which integrate with sphinx-needs to provide additional functionality.
+
+.. todo::
+
+    - Tracking progress
+        - needgantt ...
```
Comment (chrisjsewell) [inline docs/tutorial.rst:343]:
```suggestion
        - mainly to introduce needgantt
```

**Review by danwos — CHANGES_REQUESTED** [review]
Thanks for the changes.
Haven't done a full review yet, but I noticed that "useblocks" is not written always in the same way. 
Sometimes it's "UseBlocks"  or "Useblocks".
But it should be in all cases "useblocks" :)

**Comment by danwos** [comment]
It looks like the upper border is missing on need objects:
image

File: docs/directives/index.rst [hunk 2]
```diff
@@ -1,22 +1,32 @@
 Directives
 ==========
 
+Directives for creating and modifying needs:
+
 .. toctree::
+   :caption: Need creation
```
Comment (danwos) [inline docs/directives/index.rst:7]: I would remove the caption, or set it as sub-title in front of "Directives for creating and modifying needs:"

Currently, it looks like we have two introductions to the toc table:
image

File: docs/_static/flow_chart.svg
Comment (danwos) [inline docs/_static/flow_chart.svg:1]: This file looks damaged.
Downloading and opening this gives me this error:
image

File: docs/filter.rst [hunk 3]
```diff
@@ -472,3 +472,49 @@ Add specific text with this option or add no text to display nothing. The defaul
...
+      .. test:: Test for XY
+         :status: implemented
+         :tags: test; user_interface; python27
+         :links: OWN_ID_123; IMPL_01
+
+         This test checks :need:`IMPL_01` for :need:`OWN_ID_123` inside a
+         Python 2.7 environment.
+
+
+.. need-example:: Filter result as table*
```
Comment (danwos) [inline docs/filter.rst:511]: I guess the `*` is not needed

File: docs/filter.rst [hunk 4]
```diff
@@ -472,3 +472,49 @@ Add specific text with this option or add no text to display nothing. The defaul
...
+
+   .. needtable::
+      :tags: test
+      :status: implemented; open
+
+.. need-example:: Filter result as diagram
+
+   .. needflow::
```
Comment (danwos) [inline docs/filter.rst:519]: This example has no results.

File: docs/tutorial.rst [hunk 5]
```diff
@@ -0,0 +1,346 @@
...
+
+.. need-example:: Simple table
+
+    .. needtable::
+        :tags: tutorial
+        :sort: id
+        :columns: id,type,title,status
+        :style: table
+
+The default style for the table is a simple HTML table, but we can also use the ``datatables`` style to add dynamic pagination, filtering and sorting,
```
Comment (danwos) [inline docs/tutorial.rst:266]: I think the default style is the datatables style, or did this change?

**Review by danwos — CHANGES_REQUESTED** [review]
Great work, like the changes a lot.

Have 1-2 findings.

**Comment by chrisjsewell** [comment]
> It looks like the upper border is missing on need objects:
> Deactivating this CSS property make it look beautiful

can you confirm what browser (and what version) you are using @danwos; I don't see this for either firefox (mac arm64 128.0.3) or chrome (mac arm64 127.0.6533.89), and you definitely need `border-collapse` to make the inner borders show:

image

File: docs/_static/flow_chart.svg
Comment (chrisjsewell) [reply docs/_static/flow_chart.svg:1]: fixed

File: docs/filter.rst (hunk 3 above)
Comment (chrisjsewell) [reply docs/filter.rst]: fixed

**Comment by chrisjsewell** [comment]
> But it should be in all cases "useblocks" :)

fixed

**Comment by chrisjsewell** [comment]
>  I don't see this for either firefox (mac arm64 128.0.3) or chrome (mac arm64 127.0.6533.89)

Ok @danwos, this is where it gets weird.... 

- when I view the page on my MacBook monitor, it all looks fine (for Firefox/Chrome)
- BUT, when I look at the page on my external monitor, I see the same as you!

So perhaps its something to do with the screen resolution and the width of the lines 😅🤷 (my MacBook monitor has a lot better resolution)

File: docs/tutorial.rst (hunk 5 above)
Comment (chrisjsewell) [reply docs/tutorial.rst]: nope you are correct, now fixed

**Comment by chrisjsewell** [comment]
> It looks like the upper border is missing on need objects:
> Deactivating this CSS property make it look beautiful

Ok this should be fixed now; by removing the table row border. As mentioned above, your fix of removing the `border-collapse` I do not believe is a solution, because it means you cannot add "internal" borders.

I agree it would be nice to have rounded edges, but will look at that probably in a later PR

File: docs/filter.rst (hunk 4 above)
Comment (chrisjsewell) [reply docs/filter.rst:519]: fixed

File: docs/directives/index.rst (hunk 2 above)
Comment (chrisjsewell) [reply docs/directives/index.rst]: done

**Comment by danwos** [comment]
Thanks THanks for the fixes.

I detected another small border-issue:
image

Source: https://sphinx-needs--1184.org.readthedocs.build/en/1184/tutorial.html#T_CAR

The bottom left border of the sidebar is missing.
All the other borders and the standard Needs are looking fine.for the fixes.

I detected another small border-issue:
image

Source: https://sphinx-needs--1184.org.readthedocs.build/en/1184/tutorial.html#T_CAR

The bottom left border of the sidebar is missing.
All the other borders and the standard Needs are looking fine.

**Review by danwos — CHANGES_REQUESTED** [review]
see last comment

**Comment by chrisjsewell** [comment]
Ok @danwos [212d955](https://github.com/useblocks/sphinx-needs/pull/1184/commits/212d955590cfd68629faffe1c17e0d38666beabf) should fix it (and add a border-radius)

I have checked the documentation for all the themes and looks to be working for all (alabaster, sphinx_rtd_theme, furo, sphinx_immaterial, pydata_sphinx_theme)

**Review by danwos — APPROVED** [review]

**Pull request merged**